stdlib/result.gr

```diff
@@ -1,89 +1,201 @@
-        Err(e) => Err(e)
-    }
+/**
+ * If the Result is `Ok(x)`, returns `Ok(fn(x))`. Returns the unmodified `Err` otherwise.
+ *
+ * @param fn: The function to map
+ * @param result: The result to map on
+ * @returns `Ok(fn(okValue))` if the result is of the `Ok` variant, otherwise returns the unmodified `Err`
+ */
+export let map = (fn, result) => {
+  match (result) {
+    Ok(x) => Ok(fn(x)),
+    Err(e) => Err(e),
+  }
 }
 
-export let mapErr = (fn, v) => {
-    match (v) {
-        Err(e) => Err(fn(e)),
-        Ok(x)  => Ok(x)
-    }
+/**
+ * If the Result is `Err(x)`, returns `Err(fn(x))`. Returns the unmodified `Ok` otherwise.
+ *
+ * @param fn: The function to map
+ * @param result: The result to map on
+ * @returns `Err(fn(errValue))` if the result is of the `Err` variant, otherwise returns the unmodified `Ok`
+ */
+export let mapErr = (fn, result) => {
+  match (result) {
+    Err(e) => Err(fn(e)),
+    Ok(x) => Ok(x),
+  }
 }
 
-export let mapWithDefault = (fn, def, v) => {
-    match (v) {
-        Ok(a) => fn(a),
-        _     => def
-    }
+/**
+ * If the Result is `Ok(x)`, returns `fn(x)`. Returns the provided default otherwise.
+ *
+ * @param fn: The function to map
+ * @param def: The default value to return
+ * @param result: The result to map on
+ * @returns `Ok(fn(okValue))` if the result is of the `Ok` variant, otherwise returns the value given by the `def` param
+ */
+export let mapWithDefault = (fn, def, result) => {
+  match (result) {
+    Ok(a) => fn(a),
+    _ => def,
+  }
 }
 
-export let mapWithDefaultFn = (fnOk, fnErr, v) => {
-    match (v) {
-        Ok(a)  => fnOk(a),
-        Err(a) => fnErr(a)
-    }
+/**
+ * If the Result is `Ok(x)`, returns `fnOk(x)`. If the Result is `Err(y)`, returns `fnErr(y)`.
+ *
+ * @param fn: The function to map
+ * @param def: The default value to return
+ * @param result: The result to map on
+ * @returns `fnOk(okValue)` if the result is of the `Ok` variant, otherwise returns `fnErr(errValue)`
+ */
+export let mapWithDefaultFn = (fnOk, fnErr, result) => {
+  match (result) {
+    Ok(a) => fnOk(a),
+    Err(a) => fnErr(a),
+  }
 }
 
-export let or = (r1, r2) => {
-    match (r1) {
-        Ok(x) => r1,
-        _     => r2
-    }
+/**
+ * If the first Result is `Ok`, returns the first Result. Returns the second Result otherwise.
+ *
+ * @param result1: The first result
+ * @param result2: The second result
+ * @returns `result1` if result1 is of the `Ok` variant, otherwise returns `result2`
+ */
+export let or = (result1, result2) => {
+  match (result1) {
+    Ok(x) => result1,
+    _ => result2,
+  }
 }
 
-export let and = (r1, r2) => {
-    match (r1) {
-        Ok(_) => r2,
-        Err(_) => r1
-    }
+/**
+ * If the first Result is `Err`, returns the first Result. Returns the second Result otherwise.
+ *
+ * @param result1: The first result
+ * @param result2: The second result
+ * @returns `result1` if result1 is of the `Err` variant, otherwise returns `result2`
```

Comment:
```suggestion
 * @returns The second Result if both are the `Ok` variant or the first Result otherwise
```
I think this is a better way to think about the `&&`